src/atomate2/vasp/sets/mvl.py

```diff
@@ -0,0 +1,56 @@
+"""Module defining Materials Virtual Lab (MVL) VASP input set generators."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+from pymatgen.io.vasp.sets import MVLGWSet
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MVLGWSetGenerator(MVLGWSet):
+    """
+    Materials Virtual Lab GW input set generator.
+
+    To generate Materials Virtual Lab input sets for static,
+    diag and GW calculations.
+
+    Parameters
+    ----------
+    mode
+        The mode of the calculation. Options are "STATIC", "DIAG", and "GW".
+    nbands_factor
+        Multiplicative factor for NBANDS when starting from a previous calculation.
+        Choose a higher number if you are doing an LOPTICS calculation.
+    **kwargs
+        Other keyword arguments that will be passed to :obj:`VaspInputGenerator`
+    """
+
+    reciprocal_density: float = 100
+    mode: str = "STATIC"
+    copy_wavecar: bool = True
+
+    @property
+    def incar_updates(self) -> dict:
+        """Get updates to the INCAR for a Materials Virtual Lab GW calculation.
+
+        Returns
+        -------
+        dict
+            A dictionary of updates to apply.
+        """
+        updates = super().incar_updates
+        if self.mode == "DIAG":
+            # TODO: Update the values later
+            # Currently, custodian will raise VaspErrorHandler
+            # because it mistakenly thinks the calculation is not converged if NELM = 1
+            # So, I make NELM = 100 here but keep the density
+            # to be constant (ICHARG = 11).
+            updates["NELM"] = 100
```

Comment:
I don't think this is an ideal solution. I think we should submit a PR to custodian that doesn't apply the error handler if NELM = 1 and ALGO = Exact. That way we can just use the input set directly from pymatgen and we can also follow the correct VASP procedure.

Reply:
It turns out this is actually due to the issue in the `Vasprun` class in pymatgen. It mistakenly tells `custodian` that the non-self-consistent calculation is not converged. I opened a PR in which I modified two lines to jump from this condition, you may review the change here: [Fix: fixed electronic step check with algo = exact and nelm = 1](https://github.com/materialsproject/pymatgen/pull/4071)